### ISSUE TITLE
CHROME_DRIVER_VERSION usage

### ISSUE THREAD

**Issue by adamlievrouw** [SeleniumHQ/docker-selenium#245]
I trying to bring up a chrome-node with this syntax:

-e CHROME_DRIVER_VERSION=2.12

The resulting chromedriver version is still 2.21 (the default from the dockerfile). Am I calling this environmental variable incorrectly?

**Comment by jmieleiii** [comment]
Chrome node doesn't appear to have an ENV variable for setting the version

**Comment by adamlievrouw** [comment]
I'm confused, it's here no:
https://github.com/SeleniumHQ/docker-selenium/blob/365a9831aa793f7eeec00d22b0decf3328a961dc/NodeChrome/Dockerfile

#==================
# Chrome webdriver
#==================
ENV CHROME_DRIVER_VERSION 2.21
RUN wget --no-verbose -O /tmp/chromedriver_linux64.zip https://chromedriver.storage.googleapis.com/$CHROME_DRIVER_VERSION/chromedriver_linux64.zip \
  && rm -rf /opt/selenium/chromedriver \
  && unzip /tmp/chromedriver_linux64.zip -d /opt/selenium \
  && rm /tmp/chromedriver_linux64.zip \
  && mv /opt/selenium/chromedriver /opt/selenium/chromedriver-$CHROME_DRIVER_VERSION \
  && chmod 755 /opt/selenium/chromedriver-$CHROME_DRIVER_VERSION \
  && ln -fs /opt/selenium/chromedriver-$CHROME_DRIVER_VERSION /usr/bin/chromedriver

**Comment by jmieleiii** [comment]
Sorry, I confused your question with being able to set the Chrome browser version vs ChromeDriver version.

**Comment by testphreak** [comment]
Yeah, I tried the same thing and didn't work for me either. Trying to have node chrome use v2.22.

**Comment by JesseChezenko-AI** [comment]
@adamlievrouw When the image is **built** the [Dockerfile](https://github.com/SeleniumHQ/docker-selenium/blob/365a9831aa793f7eeec00d22b0decf3328a961dc/NodeChrome/Dockerfile) code is executed. ( ie. chrome driver is installed ) 

Setting an environment variable when **starting** a container, `docker run -e CHROME_DRIVER_VERSION=2.12 node-chrome`, will have no effect on the chrome driver version installed on the container. 

Note there is a pull request to update chrome driver here. https://github.com/SeleniumHQ/docker-selenium/pull/234

**Comment by F2X** [comment]
Maybe it should be a [ARG](https://docs.docker.com/engine/reference/builder/#/arg) variable and not an [ENV](https://docs.docker.com/engine/reference/builder/#/env) one?
Moreover I think we needs to find a way to offer an option to change versions on runtime, I have some issues with the couple Firefox / webdriver, which is often broken, and allowing end-user to set these values can be a plus.

**Comment by ddavison** [comment]
it definitely should be an arg, and specified by `docker build -t my-firefox-image --build-arg CHROME_DRIVER_VERSION=2.23` for example.  I've already updated it and released it in `2.53.1-americium`.

@F2X has a [pull request](https://github.com/SeleniumHQ/docker-selenium/pull/300) that was [just merged](https://github.com/SeleniumHQ/docker-selenium/commit/15d9dc0c095e1ac8f78e92c4746ee9c32ac4f13f) and will fix the same functionality with  Firefox.  This functionality will be released as `2.53.1-beryllium`

@adamlievrouw,

to fix this, you will need to build your own docker image. (these images were designed that way, anyway.)  you can build your own image with the chromedriver version you want by the following:

```bash
$ BUILD_ARGS="--build-arg CHROME_DRIVER_VERSION=<version_desired>" make build
```

or by manually invoking `docker build`
